addons/udes_stock/models/stock_picking.py

```diff
@@ -2323,3 +2346,24 @@ def reserve_stock(self):
                     break
             _logger.info("Reserving stock for picking type %r completed.", picking_type)
         return
+
+    def action_warn_picking_precondition(self):
+        """Returns a warning string for pickings if preconditions are not met."""
+        action = self.picking_type_id.u_warn_picking_precondition
```

Comment:
This will fail if self contains more than one picking. Can we add ensure_one() or loop over pickings in self, or loop over the pickings grouped by u_warn_picking_precondition please?

Reply:
Added a grouping by pickingtype

Reply:
Went with ensure_one() for now as discussed